device/alpaka/src/finding/finding_algorithm.cpp

```diff
@@ -0,0 +1,483 @@
+/** TRACCC library, part of the ACTS project (R&D line)
+ *
+ * (c) 2025 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+// Project include(s).
+#include "traccc/alpaka/finding/finding_algorithm.hpp"
+
+#include "../utils/barrier.hpp"
+#include "../utils/thread_id.hpp"
+#include "../utils/utils.hpp"
+#include "./kernels/apply_interaction.hpp"
+#include "./kernels/build_tracks.hpp"
+#include "./kernels/fill_sort_keys.hpp"
+#include "./kernels/find_tracks.hpp"
+#include "./kernels/make_barcode_sequence.hpp"
+#include "./kernels/propagate_to_next_surface.hpp"
+#include "./kernels/prune_tracks.hpp"
+#include "traccc/definitions/primitives.hpp"
+#include "traccc/definitions/qualifiers.hpp"
+#include "traccc/edm/device/sort_key.hpp"
+#include "traccc/finding/candidate_link.hpp"
+#include "traccc/geometry/detector.hpp"
+#include "traccc/utils/projections.hpp"
+
+// detray include(s).
+#include <detray/detectors/bfield.hpp>
+#include <detray/navigation/navigator.hpp>
+#include <detray/propagator/rk_stepper.hpp>
+
+// VecMem include(s).
+#include <vecmem/containers/data/vector_buffer.hpp>
+#include <vecmem/containers/data/vector_view.hpp>
+#include <vecmem/containers/device_vector.hpp>
+#include <vecmem/containers/jagged_device_vector.hpp>
+#include <vecmem/containers/vector.hpp>
+#include <vecmem/memory/unique_ptr.hpp>
+
+// Thrust include(s).
+#include <thrust/copy.h>
+#include <thrust/execution_policy.h>
+#include <thrust/fill.h>
+#include <thrust/scan.h>
+#include <thrust/sort.h>
+#include <thrust/unique.h>
+
+// System include(s).
+#include <cassert>
+#include <vector>
+
+namespace traccc::alpaka {
+
+template <typename stepper_t, typename navigator_t>
+finding_algorithm<stepper_t, navigator_t>::finding_algorithm(
+    const config_type& cfg, const traccc::memory_resource& mr,
+    vecmem::copy& copy, std::unique_ptr<const Logger> logger)
+    : messaging(std::move(logger)), m_cfg(cfg), m_mr(mr), m_copy(copy) {}
+
+template <typename stepper_t, typename navigator_t>
+track_candidate_container_types::buffer
+finding_algorithm<stepper_t, navigator_t>::operator()(
+    const typename detector_type::view_type& det_view,
+    const bfield_type& field_view,
+    const typename measurement_collection_types::view& measurements,
+    const bound_track_parameters_collection_types::buffer& seeds_buffer) const {
+
+    // Setup alpaka
+    auto devHost = ::alpaka::getDevByIdx(::alpaka::Platform<Host>{}, 0u);
+    auto devAcc = ::alpaka::getDevByIdx(::alpaka::Platform<Acc>{}, 0u);
+    auto queue = Queue{devAcc};
+    Idx threadsPerBlock = getWarpSize<Acc>() * 2;
+
+    // Copy setup
+    m_copy.setup(seeds_buffer)->ignore();
+
+#if defined(ALPAKA_ACC_GPU_CUDA_ENABLED) || defined(ALPAKA_ACC_GPU_HIP_ENABLED)
+    auto thrustExecPolicy = thrust::device;
+#else
+    auto thrustExecPolicy = thrust::host;
+#endif
```

Comment:
Why not provide these types through the same type traits that provide the vecmem type names at the moment? At which point those traits could really be moved out of the "vecmem specific" sub-namespace. 🤔